adapter/memQuota/memQuota.go

```diff
@@ -296,27 +288,25 @@ func (mq *memQuota) reapDedup() {
 	}
 }
 
-// Produce a unique key representing the cell
-func makeKey(args adapter.QuotaArgs) string {
+// Produce a unique key representing the given labels.
+func makeKey(name string, labels map[string]interface{}) string {
 	ws := keyWorkspacePool.Get().(*keyWorkspace)
 	keys := ws.keys
 	buffer := ws.buffer
 
-	// get all the label names
-	for k := range args.Labels {
+	// ensure stable order
+	for k := range labels {
 		keys = append(keys, k)
 	}
-
-	// ensure stable order
 	sort.Strings(keys)
 
-	buffer = append(buffer, []byte(args.Name)...)
+	buffer = append(buffer, []byte(name)...)
```

Comment:
I'm going to take the fifth on that question.

I've updated the code to use bytes.Buffer. Looks nice now.